src/model/operation/removeoperation.js

```diff
@@ -25,17 +26,63 @@ export default class RemoveOperation extends MoveOperation {
 	 * @param {Number} baseVersion {@link engine.model.Document#version} on which operation can be applied.
 	 */
 	constructor( position, howMany, baseVersion ) {
-		// Position in a graveyard where nodes were moved.
-		const graveyardPosition = Position.createFromParentAndOffset( position.root.document.graveyard, 0 );
+		const graveyard = position.root.document.graveyard;
 
-		super( position, howMany, graveyardPosition, baseVersion );
+		super( position, howMany, new Position( graveyard, [ graveyard.getChildCount(), 0 ] ), baseVersion );
 	}
 
+	/**
+	 * @inheritDoc
+	 */
 	get type() {
 		return 'remove';
 	}
 
 	/**
+	 * Offset of the graveyard "holder" element, in which nodes removed by this operation are stored.
+	 *
+	 * @protected
+	 * @type {Number}
+	 */
+	get _holderElementOffset() {
+		return this.targetPosition.path[ 0 ];
+	}
+
+	/**
+	 * Sets {@link engine.model.operation.RemoveOperation#_holderElementOffset}.
+	 *
+	 * @protected
+	 * @param {Number} offset
+	 */
+	set _holderElementOffset( offset ) {
+		this.targetPosition.path[ 0 ] = offset;
+	}
+
+	/**
+	 * Flag informing whether this operation should insert "holder" element (`true`) or should remove nodes
+	 * into existing "holder" element (`false`). It is `true` for each `RemoveOperation` that is the first `RemoveOperation`
+	 * in it's delta which points to given holder element.
+	 *
+	 * @protected
+	 * @type {Boolean}
+	 */
+	get _insertHolderElement() {
+		if ( this.delta ) {
+			for ( let operation of this.delta.operations ) {
+				if ( operation instanceof RemoveOperation ) {
+					if ( operation == this ) {
+						return true;
+					} else if ( operation._holderElementOffset == this._holderElementOffset ) {
```

Comment:
`operation._holderElementOffset == this._holderElementOffset` this check is unclear. What you check? Why?